preload/Makefile

```
@@ -44,6 +44,11 @@ targets := $(targets) $(D)/libsb2.$(SHLIBEXT)
 
 $(D)/libsb2.o $(D)/sb_l10n.o: $(OBJDIR)/preload/exported.h
 $(OBJDIR)/preload/exported.h $(D)/ldexportlist: $(OBJDIR)/preload/wrappers.c
+# Include both paths with $(OBJDIR) and without
+# Make doesn't seem to follow the .NOTPARALLEL rule correctly
+# without
+.NOTPARALLEL: $(OBJDIR)/preload/exported.h
+.NOTPARALLEL: preload/exported.h
 $(OBJDIR)/preload/wrappers.c: preload/interface.master preload/gen-interface.pl
```

This does not look like a right use of `.NOTPARALLEL` according to the docs.

I think the issue is caused by the way the dependencies are expressed. `exported.h` can hardly depend on `wrappers.c` when both are generated with `gen-interface.pl` based on `interface.master`. Depending on the order in which they are generated, `gen-interface.pl` may get invoked repeatedly and possibly in parallel. Maybe it would be enough to move the target `exported.h` from line 46 to line 52 (47 in the original file).

```suggestion
$(D)/ldexportlist: $(OBJDIR)/preload/wrappers.c
$(OBJDIR)/preload/exported.h $(OBJDIR)/preload/wrappers.c: preload/interface.master preload/gen-interface.pl
```

---

Hm I have to look closer first. I will check and comeback.

---

Good call. Should be better now.  While investigating also also found an earlier change related to this which wasn't fully done and is now fixed.